std/experimental/logger/asynclogger.d

```diff
@@ -0,0 +1,161 @@
+module std.experimental.logger.asynclogger;
+
+import std.experimental.logger.core;
+import std.concurrency;
+
+/** The $(D AsyncLogger) processes logs asynchronously by using another $(D Logger) implementation
+as a backend. Every data logged to this $(D AsyncLogger) will be send to separate logging thread
+in order to be logged there. It can help to improve logging performance for $(D log) caller,
+for example by removing waiting for flushing logs to a file.
+Example:
+-------------
+void main()
+{
+    import std.algorithm;
+    import std.conv;
+    import std.stdio;
+    import std.datetime;
+    import std.experimental.logger;
+
+    auto fl = new FileLogger("defaultFileLogger.log", LogLevel.all);
+    void logFileDefault()
+    {
+        fl.log("test message");
+    }
+
+    auto afl = new AsyncLogger(new FileLogger("asyncFileLogger.log", LogLevel.all));
+    void logFileAsync()
+    {
+        afl.log("test message");
+    }
+
+    auto r = benchmark!(logFileDefault, logFileAsync)(100_000);
+    writeln(map!(a => to!string(to!Duration(a)))(r[]));
+}
+-------------
+*/
+class AsyncLogger : Logger
+{
+    /// Shuntdown message
+    struct ShutdownMsg {}
+
+    import std.datetime : SysTime;
+
+    /** LogEntryMsg is an aggregation copying original $(D LogEntry)
+    but $(D (hasUnsharedAliasing!LogEntryMsg == true)).
+    This aggregation will be send to logging thread.
+    */
+    protected struct LogEntryMsg
+    {
+        /// the filename the log function was called from
+        string file;
+        /// the line number the log function was called from
+        int line;
+        /// the name of the function the log function was called from
+        string funcName;
+        /// the pretty formatted name of the function the log function was
+        /// called from
+        string prettyFuncName;
+        /// the name of the module the log message is coming from
+        string moduleName;
+        /// the $(D LogLevel) associated with the log message
+        LogLevel logLevel;
+        /// the time the message was logged
+        SysTime timestamp;
+        /// the message of the log message
+        string msg;
+    }
+
+    private static void spawnedFunc(shared Logger ptr) @trusted
+    {
+        auto l = cast(Logger)(ptr);
+        bool isRunning = true;
+        while (isRunning)
+        {
+            receive(
+                (LogEntryMsg e)
+                {
+                    LogEntry tmp = LogEntry(e.file, e.line, e.funcName, e.prettyFuncName, e.moduleName, e.logLevel, ownerTid, e.timestamp, e.msg);
+                    l.forwardMsg(tmp);
+                },
+                (ShutdownMsg msg) { isRunning = false; },
+                (Exception ex) { isRunning = false; },
+                (Variant v) {}
```

Comment:
Ignoring by default is asking for trouble, please `assert` that this doesn't happen.